pandas/core/groupby/ops.py

```diff
@@ -725,7 +726,19 @@ def _aggregate_series_pure_python(self, obj: Series, func: F):
             result[label] = res
 
         result = lib.maybe_convert_objects(result, try_float=0)
-        result = maybe_cast_result(result, obj, numeric_only=True)
+
```

Comment:
why are you doing this *here*, rather than inside maybe_cast_results itself?

Reply:
Because I'm trying to get rid of maybe_cast_result altogether.  It uses `maybe_cast_to_extension_array` whereas we should be casting yoda-style (do or do not)

Reply:
(as mentioned in the OP, id actually rather remove this chunk of code entirely)

Reply:
sure but I'd rather NOT do it in groupby at all (any casting like this), and instead push it to dtypes/cast.py this seems like going backwards.

Reply:
so you're good with just ripping this out?

Reply:
> and instead push it to dtypes/cast.py this seems like going backwards.

depends on how we think of dtypes.cast.  I think of it as "low-level helper functions related to casting" NOT "anything related to casting".  I don't want to put groupby-specific casting code in there.  (i also dont like having DataFrame.reset_index code in there)

Reply:
> > and instead push it to dtypes/cast.py this seems like going backwards.
> 
> depends on how we think of dtypes.cast. I think of it as "low-level helper functions related to casting" NOT "anything related to casting". I don't want to put groupby-specific casting code in there. (i also dont like having DataFrame.reset_index code in there)

i don't think virtually any casting code should be in groupby / frame. but we have to put it somewhere (and of course ideally there *isn't* any special casing on the class of the parent container).

so i think its the lesser of evils to keep it all together in dtypes/cast.py potentially allowing for re-use / refactoring.